source/node/Workspace.js

```diff
@@ -0,0 +1,167 @@
+     * The datasource instance for the archive
+     * @type {TextDatasource}
+     * @memberof Workspace
+     */
+    get datasource() {
+        return this._datasource;
+    }
+
+    /**
+     * The master credentials for the archive
+     * @type {Credentials}
+     * @memberof Workspace
+     */
+    get masterCredentials() {
+        return this._masterCredentials;
+    }
+
+    /**
+     * The save channel for queuing save actions
+     * @type {Channel}
+     * @memberof Workspace
+     */
+    get saveChannel() {
+        const topicID = this.archive.getID();
+        return getQueue().channel(`workspace:${topicID}`);
+    }
+
+    /**
+     * Detect whether the local archives (in memory) differ from their remote copies
+     * Fetches the remote copies from their datasources and detects differences between
+     * them and their local counterparts. Does not change/update the local items.
+     * @returns {Promise.<Boolean>} A promise that resolves with a boolean - true if
+     *      there are differences, false if there is not
+     * @memberof Workspace
+     */
+    localDiffersFromRemote() {
+        return this.datasource
+            .load(this.masterCredentials)
+            .then(history => Archive.createFromHistory(history))
+            .then(function(loadedItem) {
+                const comparator = new Comparator(this.archive, loadedItem);
+                return comparator.archivesDiffer();
+            });
+    }
+
+    /**
+     * Merge remote contents
+     * Detects differences between a local and a remote item, and merges the
+     * two copies together.
+     * @returns {Promise.<Archive>} A promise that resolves with the newly merged archive -
+     *      This archive is automatically saved over the original local copy.
+     * @memberof Workspace
+     */
+    mergeFromRemote() {
+        return this.datasource.load(this.masterCredentials).then(function(stagedArchive) {
+            const comparator = new Comparator(this.archive, stagedArchive);
+            const differences = comparator.calculateDifferences();
+            // only strip if there are multiple updates
+            const stripDestructive = differences.secondary.length > 0;
+            const newHistoryMain = stripDestructive
+                ? stripDestructiveCommands(differences.original)
+                : differences.original;
+            const newHistoryStaged = stripDestructive
+                ? stripDestructiveCommands(differences.secondary)
+                : differences.secondary;
+            const base = differences.common;
+            const newArchive = new Archive();
+            newArchive._getWestley().clear();
+            // merge all history and execute on new archive
+            base.concat(newHistoryStaged)
+                .concat(newHistoryMain)
+                .forEach(function(command) {
+                    newArchive._getWestley().execute(command);
+                });
+            this._archive = newArchive;
```

Comment:
`this` here points to the context of the `function` callback at line 106. Can we change that to arrow function so `this` here would actually point to the class?

Reply:
Yeah true! Wonder how it's been working up until now? Don't remember having it differently. Oh well, I'll change it :)